### ISSUE TITLE
Type Augmentation doesn't appear to work.

### ISSUE THREAD

**Issue by KieronWiltshire** [nextauthjs/next-auth#7021]
### Question 💬

I have my type augmentation in `src/types/next-auth.d.ts` with the following code:
```typescript
// next-auth.d.ts
import NextAuth from 'next-auth';

declare module 'next-auth' {

    interface User {
        id: string;
        username: string;
        email: string;
        name: string | null;
        accessToken: string;
    }

    interface Session {
        user: User;
    }

}
```

My `./tsconfig.json` file also looks like this:
```typescript
{
  "compilerOptions": {
    "typeRoots": [
      "src/types"
    ],
    "experimentalDecorators": true,
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ]
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx",
    "src/types/**/*.d.ts",
    ".next/types/**/*.ts"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

I am using the `app` directory in NextJS 13 and in my `src/app/layout.tsx` I have the following code:
```typescript
export default async function RootLayout({children}: PropsWithChildren) {
    const session = await getServerSession(authOptions);
    const user = undefined;
    console.log(session?.user);
    // ...
```

When I log the `session?.user` it provides me with null, then once I login, I get the following output:
```sh
{ name: null, email: 'test@example.com', image: undefined }
```

So my returned values are correct as they are being picked up in the `email` field correctly, but the other values aren't provided because they are still using the original type definitions. I can't figure out why.

### How to reproduce ☕️

I'm not sure as I've followed the docs on this one and it doesn't seem to work no matter what I try.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by krinklesaurus** [comment]
same as https://github.com/nextauthjs/next-auth/issues/7015

**Comment by KieronWiltshire** [comment]
Actually... I solved the issue. It was to do with how the data was being serialised. The user parameters weren't getting serialised in the JWT, so when it came to actually returning the session object, it wasn't able to return the data from the JWT because it didn't exist there. I have since now resolved this. However... I still get "image" returned even though I do not specify this in my augmentation module.
